code/modules/spells/targeted/equip/robesummon.dm

```diff
@@ -0,0 +1,60 @@
+/spell/targeted/equip_item/robesummon
+	name = "Summon Robes	"
+	desc = "A spell which will summon you a new set of robes."
+
+	school = "evocation"
+	charge_max = 300
+	invocation = "I PUT ON MY ROBE AND WIZARD HAT!"
+	invocation_type = SpI_SHOUT
+	range = -1
+	spell_flags = INCLUDEUSER //SELECTABLE hinders you here, since the spell has a range of 1 and only works on adjacent guys. Having the TARGETTED flag here makes it easy for your target to run away from you!
+
+	delete_old = 0 //Players shouldn't lose their hardsuits because they decided to summon some robes.
+
+	cooldown_min = 50
+
+	compatible_mobs = list(/mob/living/carbon/human)
+
+	hud_state = "wiz_robesummon"
+
+/spell/targeted/equip_item/robesummon/New()
+	..()
+
+	switch(pick("blue","red","marisa")) //Possible Robe is initially picked here so that the spell works the first time it is cast
```

Comment:
you can just have the ```..()``` of ```cast()``` happen AFTER the switch, and then you don't need this massive copypasta in ```New()```